site/source/docs/debugging/Sanitizers.rst

```diff
@@ -378,3 +372,56 @@ option ``malloc_context_size=0``, like this:
 This prevents ASan from reporting the location of memory leaks, as well as
 offer insight into where the memory for a heap-based memory error originated,
 but may provide tremendous speed ups.
+
+Comparison to ``SAFE_HEAP``
+---------------------------
+
+Emscripten provides a ``SAFE_HEAP`` mode, which can be activated by running
+``emcc`` with ``-s SAFE_HEAP``. This does several things, some of which overlap
+with sanitizers.
+
+In general, ``SAFE_HEAP`` focuses on the specific pain points that come up when
+targeting wasm. The sanitizers on the other hand focus on the specific pain points that are
+involved with using languages like C/C++. Those two sets overlap, but are not
+identical. Which you should use depends on which types of problems you are
+looking for. You may want to test with all sanitizers and with ``SAFE_HEAP``
+for maximal coverage, but you may need to build separately for each mode, since
+not all sanitizers are compatible with each other, and not all of them are
+compatible with ``SAFE_HEAP`` (because the sanitizers do some pretty radical
+things!). You will get a compiler error if there is an issue with the flags you
+passed. A reasonable set of separate test builds to do might be: ASan, UBsan,
+and ``SAFE_HEAP``.
+
+The specific things ``SAFE_HEAP`` errors on include:
+
+* **NULL pointer (address 0) reads or writes**. As mentioned earlier, this is
+  annoying in WebAssembly and JavaScript because 0 is just a normal address, so
+  you don't get an immediate segfault, which can be confusing.
+* **Unaligned reads or writes**. These work in WebAssembly, but on some platforms
+  an incorrectly-aligned read or write may be much slower, and with wasm2js
```

Comment:
Note: this sounds like something that could / should be fixed on the `wasm2js` side by using `DataView` instead.

This way, unaligned access will probably still be slower on some platforms, like mentioned above, but, at least, will work just like in native Wasm.

In V8, DataView should be on par, or even faster than TypedArrays these days, at least according to improvements made back here: https://v8.dev/blog/dataview

Reply:
(Although, I don't remember, is wasm2js the one that is supposed to produce valid asm.js? If so, then yeah, using DataView won't conform, but if it's meant to output general-purpose JS as a fallback, then it should.)

Reply:
Interesting, yeah, if DataView is fast enough then it could make sense. We should check in more browsers.

However it is bad for code size, which is a downside, so it would need to be optional.

(wasm2js is general JS, we are not focused on asm.js validation any more, so that's not an issue here.)

Reply:
> However it is bad for code size, which is a downside, so it would need to be optional.

I guess, but if it doesn't handle some code correctly, the extra code size is probably worth it.

Reply:
As an option, definitely, yes.